packages/aws-cdk-lib/aws-cloudfront-origins/lib/function-url-origin.ts

```diff
@@ -52,4 +78,61 @@ export class FunctionUrlOrigin extends cloudfront.OriginBase {
       originKeepaliveTimeout: this.props.keepaliveTimeout?.toSeconds(),
     };
   }
-}
\ No newline at end of file
+}
+
+/**
+ * An Origin for a Lambda Function URL with OAC.
+ */
+class FunctionUrlOriginWithOAC extends cloudfront.OriginBase {
+  private originAccessControl?: cloudfront.IOriginAccessControl;
+  private functionUrl: lambda.IFunctionUrl;
+  private readonly props: FunctionUrlOriginWithOACProps;
+
+  constructor(lambdaFunctionUrl: lambda.IFunctionUrl, props: FunctionUrlOriginWithOACProps = {}) {
+    const domainName = cdk.Fn.select(2, cdk.Fn.split('/', lambdaFunctionUrl.url));
+    super(domainName, props);
+    this.functionUrl = lambdaFunctionUrl;
+    this.props = props;
```

Comment:
I believe we need `this.originAccessControl = this.props?.originAccessControl` still to set the custom OAC. 

This is because `originAccessControlId` is being used in the `bind` method from this.originAccessControl. Hence for scenario where custom OAC is passed, I suspect it wont be able to retrieve the originAccessControlId based on the current logic. Please correct me if my understanding is incorrect.

Reply:
You are right. I fixed it.